Sources/Bow/Typeclasses/Decidable.swift

```diff
@@ -0,0 +1,247 @@
+/// Decidable is a typeclass modeling contravariant decision. Decidable is the contravariant version of Alternative.
+///
+/// Decidable basically states: Given a Kind<F, A> and a Kind<F, B> and a way to turn Z into either A or B it gives you a Kind<F, Z>.
+public protocol Decidable: Divisible {
```

Comment:
This file and pbxproj are the only changes in this PR, rest are from `452-add-divisible-type-class`